### PR TITLE
Add group argument for get_rank and get_world_size

### PR DESCRIPTION
**Pull request by LiYuRio**
### PR types

New features

### PR changes

APIs

### Describe

Add group arguments for get_rank and get_world_size. Now you can get world_size and rank for specific process group.

给get_rank和get_world_size增加group参数，这样可以获取某个指定的进程组中对应的rank和world_size。
中文文档修改，https://github.com/PaddlePaddle/docs/pull/5252

### CODE REVIEW

File: python/paddle/distributed/collective.py [hunk 1]
```diff
@@ -88,15 +88,15 @@ class Group():
     """
 
     def __init__(self, rank, rank_num, id=0, ranks=[], pg=None, name=None):
-        self.rank = rank
+        self._rank = rank
```
Comment (gongweibao) [inline python/paddle/distributed/collective.py]: What's the meaning of self._rank?

File: python/paddle/distributed/collective.py [hunk 2]
```diff
@@ -112,9 +112,17 @@ def get_group_rank(self, rank):
     def process_group(self):
         return self.pg
 
+    @property
+    def world_size(self):
+        return self.nranks
+
+    @property
+    def rank(self):
```
Comment (gongweibao) [inline python/paddle/distributed/collective.py]: What's the meaning of group's rank?

**Review by gongweibao — CHANGES_REQUESTED** [review]

File: python/paddle/distributed/collective.py (hunk 1 above)
Comment (LiYuRio) [reply python/paddle/distributed/collective.py]: removed

File: python/paddle/distributed/collective.py (hunk 2 above)
Comment (LiYuRio) [reply python/paddle/distributed/collective.py]: removed

File: python/paddle/distributed/parallel.py [hunk 3]
```diff
@@ -359,15 +360,21 @@ def train():
...
-def get_rank():
+def get_rank(group=None):
     """
-    Returns the rank of current trainer.
+    Returns the rank of current trainer in the given group, ranks are consecutive integers in [0, ``world_size``).
+    If none of the group is given, the global environment will be used as default.
 
-    Its value is equal to the value of the environment variable ``PADDLE_TRAINER_ID`` . 
-    The default value is 0.
+    The value is equal to ``PADDLE_TRAINER_ID`` in global environment.
```
Comment (gongweibao) [inline python/paddle/distributed/parallel.py]: Don't export this to users:`The value is equal to ``PADDLE_TRAINER_ID`` in global environment.`

File: python/paddle/distributed/parallel.py [hunk 4]
```diff
@@ -359,15 +360,21 @@ def train():
     return group
 
 
-def get_rank():
+def get_rank(group=None):
     """
-    Returns the rank of current trainer.
+    Returns the rank of current trainer in the given group, ranks are consecutive integers in [0, ``world_size``).
+    If none of the group is given, the global environment will be used as default.
```
Comment (gongweibao) [inline python/paddle/distributed/parallel.py]: `environment`=>group

File: python/paddle/distributed/parallel.py [hunk 5]
```diff
@@ -402,4 +423,14 @@ def get_world_size():
...
             # The world_size is 4
     """
+    if in_dygraph_mode():
+        # NOTE(liyurui): We need to ensure that this API is also correct in
+        # non distributed mode.
+        if collective.is_initialized():
+            group = collective._get_default_group() if group is None else group
+            return group.world_size
+        else:
+            return 1
```
Comment (gongweibao) [inline python/paddle/distributed/parallel.py]: Raise exception.

**Review by gongweibao — CHANGES_REQUESTED** [review]
Maintain the interfaces semantics

File: python/paddle/distributed/parallel.py (hunk 4 above)
Comment (LiYuRio) [reply python/paddle/distributed/parallel.py]: 已修改

File: python/paddle/distributed/parallel.py (hunk 3 above)
Comment (LiYuRio) [reply python/paddle/distributed/parallel.py]: 已修改

File: python/paddle/distributed/parallel.py (hunk 5 above)
Comment (LiYuRio) [reply python/paddle/distributed/parallel.py]: This need to add is_initialize() api before raise exception

File: python/paddle/distributed/parallel.py [hunk 6]
```diff
@@ -359,47 +359,65 @@ def train():
...
+def get_rank(group=None):
     """
-    Returns the rank of current trainer.
+    Returns the rank of current trainer in the given group, ranks are consecutive integers in [0, ``world_size``).
+    If none of the group is given, the global group will be used as default.
 
-    Its value is equal to the value of the environment variable ``PADDLE_TRAINER_ID`` . 
-    The default value is 0.
+    Args:
+        group (Group, optional): The communication group you want to get rank of current trainer, use global group as default.
```
Comment (gongweibao) [inline python/paddle/distributed/parallel.py]: `use global group as default if group is None.`

**Review by gongweibao — CHANGES_REQUESTED** [review]
A comment.

File: python/paddle/distributed/parallel.py (hunk 6 above)
Comment (LiYuRio) [reply python/paddle/distributed/parallel.py]: done

**Review by gongweibao — APPROVED** [review]
LGTM

**Review by liuTINA0907 — APPROVED** [review]

**Review by liuTINA0907 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_world_size_and_rank PROPERTIES TIMEOUT "120")`

**Pull request merged**